### PR TITLE
[MLN-901] holds non existent teacher sets

### PR DESCRIPTION
**Pull request by gonuguntla**

### CODE REVIEW

File: app/models/teacher_set.rb [hunk 1]
```diff
@@ -198,6 +198,8 @@ def self.create_or_update_teacher_set(req_body)
       # If bib request-body has suppressed value as true then delete teacher-set from database and elastic-search.
       teacher_set = self.delete_teacher_set(bib_id, req_body["suppressed"])
       if teacher_set.destroyed?
+        LogWrapper.log('INFO', {message: "message: #{BIB_RECORD_SUPPRESSED_REMOVED_FROM_MLN[:msg]}",
+                       method: __method__, bib_id: bib_id, teacher_set_id: teacher_set.id, suppressed: req_body["suppressed"]})
```
Comment (thisisstephenbetts) [inline app/models/teacher_set.rb:202]: I assume `teacher_set.id` still has a value even after it's been deleted? Just wondering if you needed to save `ts_id` before doing `delete_teacher_set`

File: test/models/hold_change_test.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    @hold_changes1.status = 'new'
+    resp = @hold_changes1.send_change_status_email
+    assert_nil(resp)
+  end
+
+
+  test 'send hold teacher_set deleted email' do
+    @hold_changes1.status = 'closed'
+    resp = @hold_changes1.send_teacher_set_deleted_email
+    assert_equal("Order closed | Teacher Set you requested has been deleted.", resp.subject)
```
Comment (thisisstephenbetts) [inline test/models/hold_change_test.rb:39]: Could you just change this to "Order closed | The Teacher Set you requested  has been deleted", so adding 'The" and omitting the '.' at the end of the subject

**Review by thisisstephenbetts — CHANGES_REQUESTED** [review]
Hi Deepika, some very minor changes requested, and a questions. But really good work. Thanks!
Stephen.

File: app/admin/hold_changes.rb [hunk 3]
```diff
@@ -22,9 +22,11 @@ def create
       # Update teacher-set available copies while cancel or closed the hold.
       if [CANCELLED, CLOSED].include?(params[:hold_change]['status'])
         hold = Hold.find(params[:hold_change]['hold_id'])
-        ts = TeacherSet.find(hold.teacher_set_id)
-        hold.teacher_set.available_copies = hold.teacher_set.available_copies.to_i + ts.holds_count_for_user(current_user, hold.id).to_i
-        hold.teacher_set.save!
+        teacher_set = TeacherSet.where(id: hold.teacher_set_id).first
```
[the remark this answers is not in the document]
Comment (gonuguntla) [reply app/admin/hold_changes.rb:25]: @thisisstephenbetts  this is not hold creation. We have one more table called HoldChange. Every status change time creating a hold_change  with latest status.

File: app/models/teacher_set.rb (hunk 1 above)
Comment (gonuguntla) [reply app/models/teacher_set.rb:202]: @thisisstephenbetts  self.delete_teacher_set(bib_id, req_body["suppressed"]) returns teacher_set object. I already saved teacher_set object before deletion.

``` # Delete teacher-set record from db and elastic search.
  def self.delete_teacher_set(bib_id, suppressed=false)
    # Get teacher-set record by bib_id
    teacher_set = self.get_teacher_set_by_bnumber(bib_id)
    unless teacher_set.present?
      if suppressed
        raise SuppressedBibRecordException.new(BIB_RECORD_SUPPRESSED_NOT_ADDED_TO_MLN[:code],BIB_RECORD_SUPPRESSED_NOT_ADDED_TO_MLN[:msg])
      else
        raise BibRecordNotFoundException.new(BIB_RECORD_NOT_FOUND[:code],BIB_RECORD_NOT_FOUND[:msg])
      end
    end
    
    # Delete teacher-set record
    resp = teacher_set.destroy
    # Feature flag: 'teacherset.data.from.elasticsearch.enabled = true'.
    # If feature flag is enabled delete data from elasticsearch.
    if MlnConfigurationController.new.feature_flag_config('teacherset.data.from.elasticsearch.enabled')
      # After deletion of teacherset data from db than delete teacherset doc from elastic search
      teacher_set.delete_teacherset_record_from_es if resp.destroyed?
    end
    teacher_set
  end```

File: app/admin/hold_changes.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (thisisstephenbetts) [reply app/admin/hold_changes.rb:25]: Thank you! That's clear.

File: app/models/teacher_set.rb (hunk 1 above)
Comment (thisisstephenbetts) [reply app/models/teacher_set.rb:202]: Great!

**Review by thisisstephenbetts — APPROVED** [review]
Thanks Deepika!

**Pull request merged**